client/src/main/java/com/battleships/windowScalling/SceneSizeChangeListener.java

```diff
@@ -0,0 +1,42 @@
+package com.battleships.windowScalling;
+
+
+import javafx.beans.value.ChangeListener;
+import javafx.beans.value.ObservableValue;
+import javafx.scene.Parent;
+import javafx.scene.Scene;
+import javafx.scene.transform.Scale;
+
+class SceneSizeChangeListener implements ChangeListener<Number> {
+    private final Scene scene;
+    private final ScalingParams scalingParams;
+    private final Parent contentPane;
+
+    public SceneSizeChangeListener(Scene scene, ScalingParams scalingParams, Parent contentPane) {
+        this.scene = scene;
+        this.scalingParams = scalingParams;
+        this.contentPane = contentPane;
+    }
+
+    @Override
+    public void changed(ObservableValue<? extends Number> observableValue, Number oldValue, Number newValue) {
+        double scaleFactor = calculateScaleFactor(scene.getWidth(), scene.getHeight());
+        if (scaleFactor >= 1) {
```

Comment:
magic number

Reply:
expression extracted to descriptive method :)